aspnet/fundamentals/logging/sample/src/LoggingSample/RequestLoggerMiddleware.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Http;
+using Microsoft.Framework.Logging;
+using Microsoft.Framework.Logging.Internal;
+
+namespace LoggingSample
+{
+    public class RequestLoggerMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ILogger _logger;
+        private readonly static Random _random = new Random();
+
+        public RequestLoggerMiddleware(RequestDelegate next, 
+            ILoggerFactory loggerFactory)
+        {
+            _next = next;
+            _logger = loggerFactory.CreateLogger<RequestLoggerMiddleware>();
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            var eventId = _random.Next();
```

Comment:
To add some clarity to `eventId`...

Suppose you're writing logging for the `TemplateRoute` class - you'd want to log matches and non-matches. Since you're in routing, your source is called `Microsoft.AspNet.Routing.TemplateRoute`. You'd log 'success' with `eventId = 10001` and 'failed' with `eventId = 10002`. This way a programmatic consumer can distinguish two statements with the same source, but different meaning.